### ISSUE TITLE
Azure Kubernetes Services - Ensure that the MaxSurge is equal or inferior to 33% if not set to Default

### ISSUE THREAD

**Issue by tdefise** [bridgecrewio/checkov#5663]
**Describe the issue**
It seems that there is no check in relation to the "MaxSurge" value in regards of Azure Kubernetes Services for Terraform.

**Examples**

````hcl
resource "azurerm_kubernetes_cluster" "cluster" {
  name                      = "example"
  location                  = var.location
  ...
  
  default_node_pool {
    name                         = "defaultpool"
    vm_size                      = var.pool_vm_size
    enable_auto_scaling          = true
    upgrade_settings {
      max_surge = "33%"
    }
  }

resource "azurerm_kubernetes_cluster_node_pool" "example" {
  name                       = "example"
  kubernetes_cluster_id = azurerm_kubernetes_cluster.cluster.id
  ...
  upgrade_settings {
    max_surge = "33%"
  }
}

````

**Version (please complete the following information):**
N/A

**Additional context**

By default, AKS configures upgrades to surge with one extra node. A default value of one for the max surge settings enables AKS to minimize workload disruption by creating an extra node before the cordon/drain of existing applications to replace an older versioned node. The max surge value can be customized per node pool to enable a trade-off between upgrade speed and upgrade disruption. When the max surge value is increased, the upgrade process completes faster. If a larger value is set for max surge, there may be disruptions during the upgrade process.

For example, a max surge value of 100% provides the fastest possible upgrade process (doubling the node count) but also causes all nodes in the node pool to be drained simultaneously. It may be possible to set a higher value such as this for testing environments. 
**For production node pools, Microsoft recommends a max_surge setting of 33%**.

TLDR: This is a setting to adjust the number of node still serving during an upgrade, and so can impact how an application can still serve during an upgrade.

Source: https://learn.microsoft.com/en-us/azure/aks/upgrade-cluster?tabs=azure-cli#customize-node-surge-upgrade

**Comment by tdefise** [comment]
@gruebel if it get accepted, I would love to work on it, but it's not gonna as simple as the other checks.
The variable is an string, but can be a int as well, as it can be either:
- a percentage ("33%") as a string
- a count (5) as an int

So we will need to check the type first (with function 'type()') in Python and then if it's an string, replace the '%', by nothing and converted to an integer
That would be my approach, but I'm open to any other

**Comment by ChananM** [comment]
Hi @tdefise, thank you for reaching out.
Thank you for your contribution. I have commented on your PR. Please review and fix if necessary and we will merge the new check.
